bark/world/evaluation/ltl/label_functions/rightmost_lane_label_function.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2020 fortiss GmbH
+//
+// Authors: Julian Bernhard, Klemens Esterle, Patrick Hart and
+// Tobias Kessler
+//
+// This work is licensed under the terms of the MIT license.
+// For a copy, see <https://opensource.org/licenses/MIT>.
+
+#include "bark/world/evaluation/ltl/label_functions/rightmost_lane_label_function.hpp"
+
+#include "bark/world/observed_world.hpp"
+
+namespace bark {
+namespace world {
+namespace evaluation {
+
+RightmostLaneLabelFunction::RightmostLaneLabelFunction(
+    const std::string& label_str, const double distance_thres)
+    : MultiAgentLabelFunction(label_str), distance_thres_(distance_thres) {}
+
+bool RightmostLaneLabelFunction::EvaluateAgent(
+    const world::ObservedWorld& observed_world,
+    const AgentPtr& other_agent) const {
+  const auto& pos = other_agent->GetCurrentPosition();
+
+  const auto right_lc =
+      other_agent->GetRoadCorridor()->GetLeftRightLaneCorridor(pos).second;
+  bool is_rightmost_lane;
+  if (right_lc) {
```

Comment:
this seems like a lot of code duplication. I do not fully grasp the label infrastructure, but it seems to take up a lot of build time. Maybe we should quantify this in the future?